Sudoku/sudoku.js

```diff
@@ -1,18 +1,63 @@
-//
-// This is only a SKELETON file for the "Sudoku" exercise. It's been provided as a
-// convenience to get you started writing code faster.
-//
+const {SQUARES, BOARD_LENGTH, NUMBERS_PER_SQUARE} = require('./constants')
+const BuildFunctions = require('./buildFunctions')
 
-var Sudoku = function(board) {
-  
-};
+module.exports = class Sudoku{
+  constructor(board){
+    this.board = board
+    this.solved = false
 
-Sudoku.prototype.isSolved = function() {
+    // Build object from squares location map
+// TODO: Will be use this? :
+// this.squaresObj = this.buildSquareContainer('object')
+    this.squaresArr = BuildFunctions.buildSquareContainer('array', this.board)
+    this.rowsArr = BuildFunctions.buildRowContainer(this.board)
+    this.columnsArr = BuildFunctions.buildColumnContainer(this.board)
 
-};
+    // Destructuring, OOP style | http://stackoverflow.com/questions/32413025/es6-destructuring-in-class-constructor
+    Object.assign( this, BuildFunctions.buildNumbersUsedNeeded(this.squaresArr) )
+  }
 
-Sudoku.prototype.solve = function() {
+  isSolved() {
+    let error = this.errorChecking()
+    if(error) return error
 
-};
+    const regex = /\./
+    const condition = this.board.match(regex) !== null
+    if(condition) return false
 
-module.exports = Sudoku;
+    return true
+  }
+
+  errorChecking(){
+    return require('./errorChecking')
+      (this.board
+     , this.squaresArr
```

Comment:
Okay, I'm not sure I fully understand what's going on here.  Are you creating an instance of the error checking file with all of these variables to initialize it?  I don't see anything in ./errorChecking that explicitly takes these values.  It's just a bunch of functions.